core/src/providers/anthropic.rs

```diff
@@ -0,0 +1,381 @@
+                self.api_key = Some(api_key.clone());
+            }
+            None => match tokio::task::spawn_blocking(|| std::env::var("ANTHROPIC_API_KEY")).await?
+            {
+                Ok(key) => {
+                    self.api_key = Some(key);
+                }
+                Err(_) => Err(anyhow!(
+                    "Credentials or environment variable `ANTHROPIC_API_KEY` is not set."
+                ))?,
+            },
+        }
+        Ok(())
+    }
+
+    fn context_size(&self) -> usize {
+        8000
+    }
+
+    async fn generate(
+        &self,
+        prompt: &str,
+        max_tokens: Option<i32>,
+        temperature: f32,
+        n: usize,
+        stop: &Vec<String>,
+        _frequency_penalty: Option<f32>,
+        _presence_penalty: Option<f32>,
+        top_p: Option<f32>,
+        top_logprobs: Option<i32>,
+        _extras: Option<Value>,
+        _event_sender: Option<UnboundedSender<Value>>,
+    ) -> Result<LLMGeneration> {
+        assert!(self.api_key.is_some());
+        assert!(n > 0);
+
+        // anthropic only supports generating one sample at a time
+        // so we loop here and make n API calls
+        let mut completions: Vec<Tokens> = vec![];
+        for _ in 0..n {
+            let response = self
+                .generate(
+                    prompt,
+                    match max_tokens {
+                        Some(m) => m,
+                        None => 256,
+                    },
+                    temperature,
+                    match top_p {
+                        Some(p) => p,
+                        None => 1.0,
+                    },
+                    match top_logprobs {
+                        Some(k) => k,
+                        None => 0,
+                    },
+                    stop,
+                )
+                .await?;
+
+            completions.push(Tokens {
+                // Anthropic only return the text
+                text: response.completion.clone(),
+                tokens: None,
+                logprobs: None,
+                top_logprobs: None,
+            });
+        }
+
+        let llm_generation = LLMGeneration {
+            created: utils::now(),
+            completions: completions,
+            provider: ProviderID::Anthropic.to_string(),
+            model: self.id.clone(),
+            prompt: Tokens {
+                text: prompt.to_string(),
+                tokens: None,
+                logprobs: None,
+                top_logprobs: None,
+            },
+        };
+
+        Ok(llm_generation)
+    }
+
+    async fn encode(&self, _text: &str) -> Result<Vec<usize>> {
+        Err(anyhow!(
+            "Encode/Decode not implemented for provider `anthropic`"
+        ))
+    }
+
+    async fn decode(&self, _tokens: Vec<usize>) -> Result<String> {
+        Err(anyhow!(
+            "Encode/Decode not implemented for provider `anthropic`"
+        ))
+    }
+
+    async fn chat(
```

Comment:
OOC, I haven't checked their API docs. Nothing like a chat-like format (given how Claude is geared towards chats?)

Reply:
No, they have only a single API endpoint as far as I can tell and it generates completions, not chat.
The Claude playground is a chat UI:
<img width="1234" alt="Screenshot 2023-04-17 at 15 00 56" src="https://user-images.githubusercontent.com/14199823/232492316-109e33bd-970a-4377-a6d2-9385610f6279.png">

They have "Edit Chat" at the top right, which reveals how it's done under the hood:
<img width="1243" alt="Screenshot 2023-04-17 at 15 02 43" src="https://user-images.githubusercontent.com/14199823/232492483-41117ddd-f54f-4bee-822e-8061f1e82c2a.png">